compiler/rustc_expand/src/expand.rs

```diff
@@ -555,6 +555,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
                     sess: &self.cx.sess,
                     features: self.cx.ecfg.features,
                     config_tokens: false,
+                    lint_node_id: self.cx.current_expansion.lint_node_id,
```

Comment:
Ok, if `lint_node_id` is initialized like this, then it's quite possible that it's always `CRATE_NODE_ID` (or maybe even an unrelated node).

`lint_node_id` shouldn't be a part of `StripUnconfigured`, it should be rather be passed on use from `fn expand_cfg_true`/`fn expand_cfg_attr`.

Reply:
Done, this has enable one more case to pass: `#[allow(unexpected_cfgs)] mod a { ... }` however the code is now "infected" with `lint_node_id`.

Pushed as https://github.com/rust-lang/rust/pull/94433/commits/d76a1808823e9ae81064aef899a457f1d253832a so you clearly see the difference with the previous approach.

Reply:
>however the code is now "infected" with `lint_node_id`

Alternatively, `StripUnconfigured`s can be created on the fly instead of being kept in `InvocationCollector`, it's just a few references aggregated together, not something expensive.

Reply:
Brilliant! Why didn't I think of that before! :smile: 
Pushed and squashed.